app-modules/assistant/src/Observers/AiAssistantObserver.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Assistant\Observers;
+
+use AdvisingApp\Assistant\Models\AiAssistant;
+use AdvisingApp\Assistant\Actions\UpdateAiAssistant;
+use AdvisingApp\Assistant\Actions\CreateCustomAiAssistant;
+use AdvisingApp\Assistant\DataTransferObjects\AiAssistantUpdateData;
+
+class AiAssistantObserver
+{
+    public function created(AiAssistant $assistant): void
+    {
+        resolve(CreateCustomAiAssistant::class)->create($assistant);
+    }
+
+    public function updated(AiAssistant $assistant): void
+    {
+        resolve(UpdateAiAssistant::class)->from(
+            assistantId: $assistant->assistant_id,
+            data: new AiAssistantUpdateData(
+                name: $assistant->name,
+                description: $assistant->description,
+                instructions: "{$assistant->instructions} {$assistant->knowledge}",
+                model: $assistant->model,
+            )
+        );
```

Comment:
As above, if this fails, we will be left with an updated model in our system, but the assistant will not be updated in the external service.

Should we consider reversing the steps and doing these on updating and creating? To gracefully fail if it is not created or updated in the external service?